src/pages/EditQuestion.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { DataGrid } from '@material-ui/data-grid';
+
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+
+import LinkIcon from '@material-ui/icons/Link';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+import ShareIcon from '@material-ui/icons/Share';
+
+import API from '@/API';
+import UserContext from '@/user';
+import ClipboardButton from '@/components/shared/ClipboardButton';
+import NewDownloadButton from '@/components/shared/NewDownloadButton';
+
+import { formatDateTimeNicely } from '@/utils/cache';
+
+export default function EditQuestion({ question, onUpdated }) {
+  const history = useHistory();
+
+  const fullLocation = `${location.origin}/question/${question.id}`;
+
+  const [newQuestion, updateNewQuestion] = useState(question);
+  const user = useContext(UserContext);
+
+  function save() {
+    API.updateQuestion(newQuestion, user.id_token);
+    onUpdated();
+  }
+  async function handleDelete() {
+    await API.deleteQuestion(question.id, user.id_token);
+    onUpdated();
+  }
+
+  const [historicAnswers, updateHistoricAnswers] = useState([]);
+
+  async function fetchHistoricAnswers() {
+    const response = await API.getAnswersByQuestion(question.id, user.id_token);
+    const answers = response;
+
+    // Spread metadata object so that we don't have nested keys
+    // when rendering in DataGrid
+    updateHistoricAnswers(answers.map((a) => ({ ...a.metadata, ...a })));
+  }
+
+  useEffect(() => { fetchHistoricAnswers(); }, [user]);
+
+  return (
+    <Box mx={1}>
+
+      <Box my={3} width={1 / 2}>
+        <TextField
+          fullWidth
+          value={newQuestion.metadata.name}
```

Comment:
This shouldn't happen outside of testing - we should consider the name field on metadata field "required". I think it makes sense not to fix this here for that reason - it should throw a console error if the name field is undefined.

Reply:
What about for messages that are uploaded from simple viewer? We might be able to guarantee a question name when asking a new question, but not for otherwise valid uploaded messages.

Reply:
I think for right now, we need to not expect question name to always exist, as it's not imperative for anything outside of this interface.

Reply:
We decided that for the Question List, questions asked or uploaded should have some sort of question name. Uploaded messages from Simple Viewer will have a standard name of 'Simple Viewer Question - <date and time>'.